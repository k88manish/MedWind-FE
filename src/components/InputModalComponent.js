import React from "react";
import { Modal, Button, Form } from "react-bootstrap";

class InputModalComponent extends React.PureComponent {
  state = {
    place: ""
  };

  onPlaceChange = e => {
    this.setState({ place: e.target.value });
  };

  onConfirm = () => {
    const { handleConfirm } = this.props;
    const { place } = this.state;
    handleConfirm(place);
    this.setState({ place: "" });
  };

  onClose = () => {
    const { handleClose } = this.props;
    this.setState({ place: "" });
    handleClose();
  };

  render() {
    const { handleClose, show, title } = this.props;
    return (
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>{title}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <Form.Control type="text" onChange={this.onPlaceChange} value={this.state.place} />
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button className="confirm-btn" variant="primary" onClick={this.onConfirm}>
            Confirm
          </Button>
          <Button className="close-btn" variant="secondary" onClick={this.onClose}>
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}

export default InputModalComponent;
