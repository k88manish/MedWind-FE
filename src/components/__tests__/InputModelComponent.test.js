import React from "react";
import { shallow } from "enzyme";
import InputModalComponent from "../InputModalComponent";

describe("InputModalComponent testing", () => {
  const handleCloseMock = jest.fn();
  const handleConfirmMock = jest.fn();
  const props = {
    show: true,
    handleClose: handleCloseMock,
    handleConfirm: handleConfirmMock,
    message: "test",
    title: "test title"
  };
  const component = shallow(<InputModalComponent {...props} />);

  it("should render properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call handleClose function when close button clicked", () => {
    const btn = component.find(".close-btn");
    btn.simulate("click");
    expect(handleCloseMock).toHaveBeenCalledTimes(1);
  });

  it("should call handleConfirm function when confirm button clicked", () => {
    const btn = component.find(".confirm-btn");
    btn.simulate("click");
    expect(handleConfirmMock).toHaveBeenCalledTimes(1);
  });
});
