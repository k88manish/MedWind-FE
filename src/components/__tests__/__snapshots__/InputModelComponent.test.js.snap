// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputModalComponent testing should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputModalComponent
    handleClose={[MockFunction]}
    handleConfirm={[MockFunction]}
    message="test"
    show={true}
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          closeButton={true}
          closeLabel="Close"
        >
          <ForwardRef>
            test title
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            as="form"
            inline={false}
          >
            <ForwardRef
              as="input"
              onChange={[Function]}
              type="text"
              value=""
            />
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            active={false}
            className="confirm-btn"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Confirm
          </ForwardRef>
          <ForwardRef
            active={false}
            className="close-btn"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="secondary"
          >
            Close
          </ForwardRef>
        </ForwardRef>,
      ],
      "onHide": [MockFunction],
      "show": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            test title
          </ForwardRef>,
          "closeButton": true,
          "closeLabel": "Close",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test title",
          },
          "ref": null,
          "rendered": "test title",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalTitle",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="form"
            inline={false}
          >
            <ForwardRef
              as="input"
              onChange={[Function]}
              type="text"
              value=""
            />
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "form",
            "children": <ForwardRef
              as="input"
              onChange={[Function]}
              type="text"
              value=""
            />,
            "inline": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "input",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                  "type": "valid",
                },
                "displayName": "Feedback",
                "propTypes": Object {
                  "as": [Function],
                  "type": [Function],
                },
                "render": [Function],
              },
              "defaultProps": Object {
                "as": "input",
              },
              "displayName": "FormControl",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Check": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Input": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckInput",
                "render": [Function],
              },
              "Label": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "type": "checkbox",
                },
                "displayName": "FormCheckLabel",
                "render": [Function],
              },
              "defaultProps": Object {
                "disabled": false,
                "inline": false,
                "isInvalid": false,
                "isValid": false,
                "title": "",
                "type": "checkbox",
              },
              "displayName": "FormCheck",
              "render": [Function],
            },
            "Control": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                  "type": "valid",
                },
                "displayName": "Feedback",
                "propTypes": Object {
                  "as": [Function],
                  "type": [Function],
                },
                "render": [Function],
              },
              "defaultProps": Object {
                "as": "input",
              },
              "displayName": "FormControl",
              "render": [Function],
            },
            "Group": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "FormGroup",
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "column": false,
                "srOnly": false,
              },
              "displayName": "FormLabel",
              "render": [Function],
            },
            "Row": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "FormRow",
              "render": [Function],
            },
            "Text": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "small",
              },
              "displayName": "FormText",
              "render": [Function],
            },
            "defaultProps": Object {
              "as": "form",
              "inline": false,
            },
            "displayName": "Form",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              active={false}
              className="confirm-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Confirm
            </ForwardRef>,
            <ForwardRef
              active={false}
              className="close-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              Close
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "Confirm",
              "className": "confirm-btn",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "variant": "primary",
            },
            "ref": null,
            "rendered": "Confirm",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "Close",
              "className": "close-btn",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "variant": "secondary",
            },
            "ref": null,
            "rendered": "Close",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalFooter",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "BACKDROP_TRANSITION_DURATION": 150,
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalBody",
        "render": [Function],
      },
      "Dialog": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "ModalDialog",
        "render": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalFooter",
        "render": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "closeButton": false,
          "closeLabel": "Close",
        },
        "displayName": "ModalHeader",
        "render": [Function],
      },
      "TRANSITION_DURATION": 300,
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalTitle",
        "render": [Function],
      },
      "defaultProps": undefined,
      "displayName": "Bootstrap(Modal)",
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            closeButton={true}
            closeLabel="Close"
          >
            <ForwardRef>
              test title
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              as="form"
              inline={false}
            >
              <ForwardRef
                as="input"
                onChange={[Function]}
                type="text"
                value=""
              />
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              active={false}
              className="confirm-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              Confirm
            </ForwardRef>
            <ForwardRef
              active={false}
              className="close-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              Close
            </ForwardRef>
          </ForwardRef>,
        ],
        "onHide": [MockFunction],
        "show": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              test title
            </ForwardRef>,
            "closeButton": true,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test title",
            },
            "ref": null,
            "rendered": "test title",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "ModalTitle",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "closeButton": false,
              "closeLabel": "Close",
            },
            "displayName": "ModalHeader",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="form"
              inline={false}
            >
              <ForwardRef
                as="input"
                onChange={[Function]}
                type="text"
                value=""
              />
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "form",
              "children": <ForwardRef
                as="input"
                onChange={[Function]}
                type="text"
                value=""
              />,
              "inline": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "input",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                    "type": "valid",
                  },
                  "displayName": "Feedback",
                  "propTypes": Object {
                    "as": [Function],
                    "type": [Function],
                  },
                  "render": [Function],
                },
                "defaultProps": Object {
                  "as": "input",
                },
                "displayName": "FormControl",
                "render": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Check": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Input": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "type": "checkbox",
                  },
                  "displayName": "FormCheckInput",
                  "render": [Function],
                },
                "Label": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "type": "checkbox",
                  },
                  "displayName": "FormCheckLabel",
                  "render": [Function],
                },
                "defaultProps": Object {
                  "disabled": false,
                  "inline": false,
                  "isInvalid": false,
                  "isValid": false,
                  "title": "",
                  "type": "checkbox",
                },
                "displayName": "FormCheck",
                "render": [Function],
              },
              "Control": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Feedback": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": "div",
                    "type": "valid",
                  },
                  "displayName": "Feedback",
                  "propTypes": Object {
                    "as": [Function],
                    "type": [Function],
                  },
                  "render": [Function],
                },
                "defaultProps": Object {
                  "as": "input",
                },
                "displayName": "FormControl",
                "render": [Function],
              },
              "Group": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "FormGroup",
                "render": [Function],
              },
              "Label": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "column": false,
                  "srOnly": false,
                },
                "displayName": "FormLabel",
                "render": [Function],
              },
              "Row": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "displayName": "FormRow",
                "render": [Function],
              },
              "Text": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "small",
                },
                "displayName": "FormText",
                "render": [Function],
              },
              "defaultProps": Object {
                "as": "form",
                "inline": false,
              },
              "displayName": "Form",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalBody",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                active={false}
                className="confirm-btn"
                disabled={false}
                onClick={[Function]}
                type="button"
                variant="primary"
              >
                Confirm
              </ForwardRef>,
              <ForwardRef
                active={false}
                className="close-btn"
                disabled={false}
                onClick={[Function]}
                type="button"
                variant="secondary"
              >
                Close
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "children": "Confirm",
                "className": "confirm-btn",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
                "variant": "primary",
              },
              "ref": null,
              "rendered": "Confirm",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "children": "Close",
                "className": "close-btn",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
                "variant": "secondary",
              },
              "ref": null,
              "rendered": "Close",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalFooter",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "BACKDROP_TRANSITION_DURATION": 150,
        "Body": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
        "Dialog": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        },
        "Footer": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalFooter",
          "render": [Function],
        },
        "Header": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
        "TRANSITION_DURATION": 300,
        "Title": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalTitle",
          "render": [Function],
        },
        "defaultProps": undefined,
        "displayName": "Bootstrap(Modal)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
